docs/dev_mode.md

```diff
@@ -41,110 +41,9 @@ DEPLOYMENT_TYPE=kubernetes
-			--dev && exit 0
-		exit 1
-	fi
-	autonomy deploy build image ${SERVICE_ID} --version ${VERSION} && exit 0
-	exit 1
-
-.ONESHELL: build-images push-images
-push-images:
-	if [ "${VERSION}" = "" ];\
-	then\
-		echo "Ensure you have exported a version to build!";\
-		exit 1
-	fi
-	autonomy deploy build image ${SERVICE_ID} --dependencies --push || (echo failed && exit 1)
-	if [ "${VERSION}" = "dev" ];\
-	then\
-		echo "building dev images!";\
-		autonomy deploy build image ${SERVICE_ID} --dev --push || (echo failed && exit 1)
-		exit 0
-	fi
-	autonomy deploy build image ${SERVICE_ID} --version ${VERSION} --prod --push || (echo failed && exit 1)
-	exit 0
-
-.PHONY: build-deploy
-build-deploy:
-	if [ "${DEPLOYMENT_TYPE}" = "" ];\
-	then\
-		echo "Please ensure you have set the environment variable 'DEPLOYMENT_TYPE'"
-		exit 1
-	fi
-	if [ "${SERVICE_ID}" = "" ];\
-	then\
-		echo "Please ensure you have set the environment variable 'SERVICE_ID'"
-		exit 1
-	fi
-	if [ "${DEPLOYMENT_KEYS}" = "" ];\
-	then\
-		echo "Please ensure you have set the environment variable 'DEPLOYMENT_KEYS'"
-		exit 1
-	fi
-	echo "Building deployment for ${DEPLOYMENT_TYPE} ${DEPLOYMENT_KEYS} ${SERVICE_ID}"
-
-	if [ "${DEPLOYMENT_TYPE}" = "kubernetes" ];\
-	then\
-		if [ "${VERSION}" = "cluster-dev" ];\
-		then\
-			autonomy deploy build deployment ${SERVICE_ID} ${DEPLOYMENT_KEYS} --kubernetes --force --dev
-			exit 0
-		fi
-		autonomy deploy build deployment ${SERVICE_ID} ${DEPLOYMENT_KEYS} --kubernetes --force
-		exit 0
-	fi
-	if [ "${VERSION}" = "dev" ];\
-	then\
-		autonomy deploy build deployment ${SERVICE_ID} ${DEPLOYMENT_KEYS} --docker --dev --force
-		exit 0
-	fi
-	autonomy deploy build deployment ${SERVICE_ID} ${DEPLOYMENT_KEYS} --docker
-
-.PHONY: run-deploy
-run-deploy:
-	if [ "${PLATFORM_STR}" = "Linux" ];\
-	then\
-		mkdir -p abci_build/persistent_data/logs
-		mkdir -p abci_build/persistent_data/venvs
-		sudo chown -R 1000:1000 -R abci_build/persistent_data/logs
-		sudo chown -R 1000:1000 -R abci_build/persistent_data/venvs
-	fi
-	if [ "${DEPLOYMENT_TYPE}" = "docker-compose" ];\
-	then\
-		cd abci_build/ &&  \
-		docker-compose up --force-recreate -t 600 --remove-orphans
-		exit 0
-	fi
-	if [ "${DEPLOYMENT_TYPE}" = "kubernetes" ];\
-	then\
-		kubectl create ns ${VERSION}|| (echo "failed to deploy to namespace already existing!" && exit 0)
-		kubectl create secret generic regcred \
-          --from-file=.dockerconfigjson=/home/$(shell whoami)/.docker/config.json \
-          --type=kubernetes.io/dockerconfigjson -n ${VERSION} || (echo "failed to create secret" && exit 1)
-		cd abci_build/ && \
-			kubectl apply -f build.yaml -n ${VERSION} && \
-			kubectl apply -f agent_keys/ -n ${VERSION} && \
-			exit 0
-	fi
-	echo "Please ensure you have set the environment variable 'DEPLOYMENT_TYPE'"
-	exit 1
+autonomy build-image ${SERVICE_ID}
+autonomy deploy build ${DEPLOYMENT_KEYS} --kubernetes --force --dev
+autonomy deploy run --build-dir <build_dir>
```

Comment:
why don't you try it yourself @dvilelaf ?

Reply:
I already tried it, and got an error while building:
`Error: Package hash was not provided.`
I then appended the service hash the SERVICE_ID but got `Error: Invalid value for '[PUBLIC_ID_OR_HASH]'`.

Reply:
Ok so please debug this fully with the help of your peers.

Reply:
here are the exact steps.

```
autonomy fetch valory/oracle_hardhat --local --service
cd oracle_hardhat
autonomy build-image
autonomy deploy build  ../generated_keys.json --force --password ${PASSWORD} --kubernetes --dev
kubectl apply -f abci_build/
kubectl apply -f abci_build/agent_keys
```
I just ran this and found an issue;

1. when deploying in dev mode in the cluster, we set the ledger address host to 
```
http://host.docker.internal:8545

It should be
```
http://hardhat:8545
```
Other than this, everything works as expected.

Reply:
Raised 2 issues about this;

https://github.com/valory-xyz/open-autonomy/issues/1336
https://github.com/valory-xyz/open-autonomy/issues/1337

Reply:
@8ball030 can you point out where in the code this url needs fixing?

Reply:
Hmmm good point, this is a service_spec variable. 

I wonder whether it would be best to go ahead have a separate service spec for the clustered hardhat?

Reply:
I don't think it's worth having a separate service definition just for this. The better approach would be to replace the hardcoded url with an env variable and then ask the user to set the env variable depending on the scenario.